check_api/src/main/java/com/google/errorprone/ErrorProneAnalyzer.java

```diff
@@ -79,15 +80,18 @@ public static ErrorProneAnalyzer createAnalyzer(
             .or(
                 Suppliers.memoize(
                     () -> {
-                      ScannerSupplier toUse =
-                          ErrorPronePlugins.loadPlugins(scannerSupplier, context);
                       ImmutableSet<String> namedCheckers =
                           epOptions.patchingOptions().namedCheckers();
-                      if (!namedCheckers.isEmpty()) {
-                        toUse = toUse.filter(bci -> namedCheckers.contains(bci.canonicalName()));
-                      } else {
-                        toUse = toUse.applyOverrides(epOptions);
-                      }
+                      ScannerSupplier toUse =
+                          ErrorPronePlugins.loadPlugins(scannerSupplier, context)
+                              .applyOverrides(epOptions)
+                              .filter(
+                                  bci -> {
+                                    String name = bci.canonicalName();
+                                    return epOptions.getSeverityMap().get(name) != Severity.OFF
+                                        && (namedCheckers.isEmpty()
+                                            || namedCheckers.contains(name));
+                                  });
                       return ErrorProneScannerTransformer.create(toUse.get());
                     }));
```

Comment:
So patching now ignores checks:
- Whose severity is `OFF`, or
- That are not explicitly enabled for patching, while other checks are.